### PR TITLE
Added feature flags for relational transform execution

### PR DESCRIPTION
**Pull request by fernst**

### CODE REVIEW

File: core-plugins/pom.xml [hunk 1]
```diff
@@ -42,15 +42,15 @@
     </dependency>
     <dependency>
       <groupId>io.cdap.cdap</groupId>
-      <artifactId>hydrator-test</artifactId>
+      <artifactId>cdap-etl-api</artifactId>
     </dependency>
```
Comment (sanjanasandeep) [inline core-plugins/pom.xml]: why this?

**Review by sanjanasandeep — APPROVED** [review]

**Review by sanjanasandeep — APPROVED** [review]

File: core-plugins/src/test/java/io/cdap/plugin/batch/aggregator/GroupByRelationalTest.java [hunk 2]
```diff
@@ -54,6 +54,7 @@ public class GroupByRelationalTest {
   @Before
   public void setUp() {
     Mockito.doReturn(engine).when(relationalTranformContext).getEngine();
+    Mockito.doReturn(true).when(relationalTranformContext).isFeatureEnabled(Mockito.anyString());
     Mockito.doReturn(Optional.of(expressionFactory)).when(engine).getExpressionFactory(Mockito.any(),
```
Comment (sanjanasandeep) [inline core-plugins/src/test/java/io/cdap/plugin/batch/aggregator/GroupByRelationalTest.java:58]: we don't need this for `dedup aggregator test`?

File: core-plugins/pom.xml (hunk 1 above)
Comment (fernst) [reply core-plugins/pom.xml]: oops, fixed

File: core-plugins/src/test/java/io/cdap/plugin/batch/aggregator/GroupByRelationalTest.java (hunk 2 above)
Comment (fernst) [reply core-plugins/src/test/java/io/cdap/plugin/batch/aggregator/GroupByRelationalTest.java:58]: We are not actually testing the `transform` method for the Dedup aggregator. We only test the logic to generate the definition.

**Comment by fernst** [comment]
Solved on the Google-Cloud side

**Pull request closed without merging**